src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,232 @@
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.team3061.drivetrain.Drivetrain;
+import frc.lib.team3061.util.RobotOdometry;
+import frc.lib.team6328.util.TunableNumber;
+import frc.robot.Field2d;
+import frc.robot.subsystems.intake.Intake;
+import java.util.function.BooleanSupplier;
+import org.littletonrobotics.junction.Logger;
+
+public class Shooter extends SubsystemBase {
+
+  private ShooterIO io;
+  private Intake intake;
+  private Alliance alliance = Alliance.Red;
+  private InterpolatingDoubleTreeMap angleTreeMap;
+  private InterpolatingDoubleTreeMap veloTreeMap;
+  private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+  private final TunableNumber topWheelVelocity = new TunableNumber("Shooter/Top Wheel Velocity", 0);
+  private final TunableNumber bottomWheelVelocity =
+      new TunableNumber("Shooter/Bottom Wheel Velocity", 0);
+  private final TunableNumber angle = new TunableNumber("Shooter/Angle", 10.0);
+
+  public BooleanSupplier inStorage = () -> false;
+
+  private boolean autoShooter = false;
+  private boolean hasNote = false;
+  private boolean isAimed = false;
+  private boolean aiming = false;
+  private boolean scoringAmp = false;
+  private boolean scoringSubwoofer = false;
+  private boolean scoringPodium = false;
+
+  private int topAtSetpointIterationCount = 0;
+  private int bottomAtSetpointIterationCount = 0;
+  private int angleAtSetpointIterationCount = 0;
+
+  public Shooter(ShooterIO io, Drivetrain drivetrain, Intake intake) {
+    this.io = io;
+    this.intake = intake;
+    this.angleTreeMap = new InterpolatingDoubleTreeMap();
+    this.veloTreeMap = new InterpolatingDoubleTreeMap();
+
+    this.autoShooter = false;
+    this.hasNote = false;
+    this.isAimed = false;
+    this.aiming = false;
+    this.scoringAmp = false;
+    this.scoringSubwoofer = false;
+    this.scoringPodium = false;
+    this.inStorage = () -> false;
+
+    if (TESTING) {
+      ShuffleboardTab tab = Shuffleboard.getTab(SUBSYSTEM_NAME);
+      tab.add(SUBSYSTEM_NAME, this);
+    }
+  }
+
+  @Override
+  public void periodic() {
+    io.updateInputs(shooterInputs);
+    Logger.processInputs(SUBSYSTEM_NAME, shooterInputs);
+    this.inStorage =
+        () -> (Math.abs(shooterInputs.angleEncoderAngleDegrees) < ShooterConstants.ANGLE_TOLERANCE);
+    // this.hasNote = intake.hasNote();
+
+    if (TESTING) {
+      io.setShooterWheelBottomVelocity(bottomWheelVelocity.get());
+      io.setShooterWheelTopVelocity(topWheelVelocity.get());
+      io.setAngle(angle.get());
+    } else if (autoShooter == false) {
+      io.setShooterWheelBottomVelocity(SHOOTER_IDLE_VELOCITY);
+      io.setShooterWheelTopVelocity(SHOOTER_IDLE_VELOCITY);
+    } else {
+      this.runAngleStateMachine();
+      io.setShooterWheelBottomVelocity(SHOOTER_IDLE_VELOCITY);
```

Comment:
if you are running the state machine, you don't want to override whatever it just did by setting the shooter velocity